presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java

```diff
@@ -184,64 +188,65 @@ public ColumnarBinaryHiveRecordCursor(RecordReader<K, BytesRefArrayWritable> rec
             types[i] = typeManager.getType(column.getTypeSignature());
             hiveTypes[i] = column.getHiveType();
 
-            if (!column.isPartitionKey()) {
+            if (!column.isPartitionKey() && !column.isHidden()) {
                 fieldInspectors[i] = rowInspector.getStructFieldRef(column.getName()).getFieldObjectInspector();
             }
 
             hiveColumnIndexes[i] = column.getHiveColumnIndex();
-            isPartitionColumn[i] = column.isPartitionKey();
+            isPrefilledColumn[i] = column.isPartitionKey() || column.isHidden();
         }
 
         // parse requested partition columns
         Map<String, HivePartitionKey> partitionKeysByName = uniqueIndex(partitionKeys, HivePartitionKey::getName);
         for (int columnIndex = 0; columnIndex < columns.size(); columnIndex++) {
             HiveColumnHandle column = columns.get(columnIndex);
-            if (column.isPartitionKey()) {
+            if (column.isPartitionKey() || column.isHidden()) {
                 HivePartitionKey partitionKey = partitionKeysByName.get(column.getName());
-                checkArgument(partitionKey != null, "Unknown partition key %s", column.getName());
+                checkArgument(partitionKey != null || column.isHidden(), "Unknown partition key %s", column.getName());
```

Comment:
I would remove this assertion. Your `getPrefilledColumnValue` has the same check.